pkg/secrets/secrets.go

```diff
@@ -187,9 +183,63 @@ func SecretMounts(filePath, mountLabel, containerWorkingDir string) ([]rspec.Mou
 
 		mounts = append(mounts, m)
 	}
+
+	// Add FIPS mode secret
+	// The filePath check is to ensure that this secret is created only once
+	// as SecretMounts is called twice for the two mounts.conf files ("OverrideMountsFile" and "DefaultMountsFile") in the respective order
+	// therefore we want to add the FIPS mode secret, if it exists, after the secrets from the DefaultMountsFile are mounted
+	_, err := os.Stat("/etc/system-fips")
+	if err == nil && filePath == DefaultMountsFile {
+		if err := addFIPSsModeSecret(&mounts, containerWorkingDir); err != nil {
+			return nil, errors.Wrapf(err, "error adding FIPS mode secret to container")
+		}
+	}
 	return mounts, nil
 }
 
+// addFIPSModeSecret creates /run/secrets/system-fips in the container
+// root filesystem if /etc/system-fips exists on hosts.
+// This enables the container to be FIPS compliant and run openssl in
+// FIPS mode as the host is also in FIPS mode.
+func addFIPSsModeSecret(mounts *[]rspec.Mount, containerWorkingDir string) error {
+	secretsDir := "/run/secrets"
```

Comment:
We're going to mount this RW into all containers, which makes me think we should probably not only have one copy?

Reply:
Also, I don't like having this dir non-configurable. For libpod, I'd like to tuck it inside our tmp directory somewhere.

Reply:
Tuck the "/run/secrets" dir inside a tmp directory or the "system-fips" file. The system-fips file needs to be in /run/secrets in the container as that is what a symlink in the image will be pointing to. If the symlink is not dangling, that means the host was in fips mode so the container should run in fips mode as well.

Reply:
`/run/secrets` should be configurable, I would like to put that in a `secrets` dir under libpod's tmp dir

Reply:
Makes sense, but this same patch is going to be vendored in to buildah and cri-o. Plus @rhatdan said to have it in `/run/secrets`. Should we make /run/secrets such that the user can't overwrite it?

Reply:
/run/secrets is standard, inside the container.  RHEL containers are hard coded to look in /run/secrets in the container.  It can not be configurable.

Reply:
@rhatdan if the user volume mounts something to "/run/secrets" using the --volume flag, it overwrites any secrets that were mounted. Should this behaviour stay or should we stop the user from being ale to overwrite "/run/secrets"?
When docker run sees a duplicate mount destination, it errors out telling the user "destination" is duplicated.

Reply:
@umohnani8 I like our behavior of not erroring, but we should probably log that a mount is being ignored if we don't add it because of overlap. I don't think we do that now.

Reply:
@mheon okay, agreed.